cub/benchmarks/bench/scan/applications/log-cdf-from-log-pdfs.cu

```diff
@@ -0,0 +1,182 @@
+ * compute sequence of logarithms of cumulative distribution function values.
+ *
+ *    log(CDF(n)) = log(\sum( PDF(k), 0 <=k <=n ))
+ *
+ * This is inclusive scan using logaddexp binary operator:
+ *    logaddexp( logpdf1, logpdf2 ) := log( exp(logpdf1) + exp(logpdf2) )
+ *         == max(logpdf1, logpdf2) + log( 1 + exp(-abs(logpdf1 - logpdf2)))
+ *
+ * The last reformulation allows avoid numerical accuracy issues
+ * caused by underflows.
+ *
+ */
+
+struct LogAddPlus
+{
+  /* Operator is commutative and associative */
+  template <typename T>
+  T __host__ __device__ operator()(T v1, T v2)
+  {
+    T max12 = cuda::maximum{}(v1, v2);
+    T min12 = cuda::minimum{}(v1, v2);
+    T exp   = cuda::std::exp(min12 - max12);
+    return max12 + cuda::std::log1p(exp);
+  }
+};
+
+template <typename T>
+struct LogPDBuilder
+{
+  T mu;
+  T norm;
+  cuda::std::size_t n;
+
+  T __host__ __device__ operator()(cuda::std::size_t i) const
+  {
+    return -mu * static_cast<T>(n - i) + norm;
+  }
+};
+
+template <typename T>
+[[nodiscard]] bool validate(const thrust::device_vector<T>& output, cudaStream_t stream)
+{
+  cudaStreamSynchronize(stream);
+
+  thrust::host_vector<T> h_output(output);
+  auto elements = h_output.size();
+  // test is designed so that last element of prefix scan sequence should be close to log(1.0) == 0.0
+  bool check = cuda::std::abs(h_output[elements - 1])
+             < cuda::std::sqrt(static_cast<T>(1 + elements)) * cuda::std::numeric_limits<T>::epsilon();
+
+  return check;
+}
+
+}; // namespace impl
+
+template <typename FloatingPointT, typename OffsetT>
+static void inclusive_scan(nvbench::state& state, nvbench::type_list<FloatingPointT, OffsetT>)
+{
+  using wrapped_init_t = cub::NullType;
+  using value_t        = FloatingPointT;
+  using input_t        = const value_t*;
+  using output_t       = value_t*;
+  using offset_t       = cub::detail::choose_offset_t<OffsetT>;
+  using op_t           = impl::LogAddPlus;
+  using accum_t        = value_t;
+
+#if !TUNE_BASE
+  using policy_t = policy_hub_t<accum_t>;
+  using dispatch_t =
+    cub::DispatchScan<input_t, output_t, op_t, wrapped_init_t, offset_t, accum_t, cub::ForceInclusive::No, policy_t>;
+#else
+  using dispatch_t =
+    cub::DispatchScan<input_t, output_t, op_t, wrapped_init_t, offset_t, accum_t, cub::ForceInclusive::No>;
+#endif
+
+  const auto elements = static_cast<std::size_t>(state.get_int64("Elements{io}"));
+  auto mu             = static_cast<value_t>(state.get_float64("Mu{io}"));
+
+  auto norm = cuda::std::log1p(-cuda::std::exp(-mu)) - cuda::std::log1p(-cuda::std::exp(-mu * elements));
+
+  thrust::device_vector<value_t> input(elements, thrust::no_init);
+
+  cudaStream_t bench_stream = state.get_cuda_stream();
+
+  cub::DeviceTransform::Transform(
+    cuda::std::make_tuple(cuda::counting_iterator(cuda::std::size_t{0})),
+    input.begin(),
+    elements,
+    impl::LogPDBuilder<value_t>{mu, norm, elements},
+    bench_stream);
```

Comment:
Suggestion: Use `thrust::tabulate`

Reply:
I decided against using `tabulate`, just because I think keeping asynchrony explicit is beautiful. Color me a romantic :)

Reply:
So just use an execution policy to make it async:
`thrust::tabulate(thrust::cuda::par_nosync, ...)`